### PR TITLE
feature :: use vuex for application state

### LINKED ISSUE #131: Use vuex to manage application state

### PR DESCRIPTION
**Pull request by adimascio**
Closes #131

### CODE REVIEW

File: playground/App.vue [hunk 1]
```diff
@@ -1,70 +1,50 @@
...
 export default class App extends Vue {
-  dataset: DataSet = {
-    headers: [{ name: 'col1' }, { name: 'col2' }, { name: 'col3' }],
-    data: [
-      [12, 13, 14],
-      [15, 19, 15],
-      [16, 20, 16],
-    ]
-  };
+  @State('pipeline') steps!: Pipeline;
```
Comment (vdestraitt) [inline playground/App.vue]: Should we remain consistent and keep the `pipeline`wording rather than `steps`?

File: tests/unit/pipeline.spec.ts [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
   it('renders steps', () => {
-    const wrapper = shallowMount(Pipeline, {
-      propsData: {
-        steps: [{ name: 'domain' }, { name: 'rename' }, { name: 'group' }],
-      },
-    });
+    const pipeline: Pipeline = [
+      { name: 'domain', domain: 'GoT' },
+      { name: 'replace', search_column: 'characters', to_replace: [['Snow', 'Targaryen']] },
+      { name: 'sort', columns: ['death'] },
```
Comment (Dalanir) [inline tests/unit/pipeline.spec.ts]: Like this test aha !

File: stories/data-viewer.js [hunk 3]
```diff
@@ -1,44 +1,41 @@
...
+        ],
+      data: [
+        ['value1', 'value2', 'value3'],
+        ['value4', 'value5', 'value6'],
+        ['value7', 'value8', 'value9'],
+        ['value10', 'value11', 'value12'],
+        ['value10', { obj: 'value14' }, null],
+      ]
+    },
+  }, [], false),
```
Comment (Dalanir) [inline stories/data-viewer.js]: Why is autobind `false` and why specify it ? It looks like it's the default value

Comment (adimascio) [reply stories/data-viewer.js]: good catch, it's a mistake.

**Pull request merged**